src/axom/inlet/YAMLReader.cpp

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file YAMLReader.cpp
+ *
+ * \brief This file contains the class implementation of the YAMLReader.
+ *******************************************************************************
+ */
+
+#include <fstream>
+
+#include "axom/inlet/YAMLReader.hpp"
+
+#include "axom/core/utilities/FileUtilities.hpp"
+#include "axom/core/utilities/StringUtilities.hpp"
+#include "axom/inlet/inlet_utils.hpp"
+
+#include "fmt/fmt.hpp"
+#include "axom/slic.hpp"
+
+// #include "conduit_relay.hpp"
+
+namespace axom
+{
+namespace inlet
+{
+YAMLReader::YAMLReader() { }
+bool YAMLReader::parseFile(const std::string& filePath)
+{
+  if(!axom::utilities::filesystem::pathExists(filePath))
+  {
+    SLIC_WARNING(fmt::format("Inlet: Given YAML input file does not exist: {0}",
+                             filePath));
+    return false;
+  }
+  m_root.load(filePath, "yaml");
```

Comment:
This is a good question - failure to parse here will result in a SLIC error (via Conduit's error handler) and subsequently program termination.  Exceptions are (to my knowledge, intentionally) avoided in Axom, so I am not sure that we have the ability to detect/handle a parse error in this context.

Reply:
That behavior is inconsistent with the documentation, which states that the function just returns false if the file can't be read. Having the program terminate instead seems quite drastic, especially for something that can be used as a library. If this is reasonably easy to change, I think it should be. Otherwise, I would document this possibility in the API.

The Axom team should chime in here.

Reply:
Depending on how Sidre is used, you should be able to catch the conduit exception here, and return false (or otherwise report the details of the error)

Reply:
The exceptions thrown by conduit's default error handler are suppressed by the SLIC error handler - I'm wondering if it would be acceptable to temporarily reset the error handler to the conduit default, catch the exception if it gets thrown, then reset to the SLIC error handler.

Reply:
We just handled something similar in a different application, but I don't recall the specifics.  
@cyrush can you please help @joshessman-llnl  with the details?

Reply:
The two functions here are likely sufficient for your needs: 
https://github.com/LLNL/axom/pull/350/files#diff-2908fac76434ec9e7431bfb3b6173f7c87ae79671faec954bf97e62d77c2b98f

Reply:
Yep, it looks like those functions were enough to implement what I had proposed.